### PR TITLE
Remove various unused fields in `naga`

### PR DESCRIPTION
**Pull request by ErichDonGubler**
**Connections**
_Link to the issues addressed by this PR, or dependent PRs in other repositories_

**Description**
_Describe what problem this is solving, and how it's solved._

**Testing**
_Explain how this change is tested._

**Checklist**

- [x] Run `cargo fmt`.
- [x] Run `cargo clippy`. If applicable, add:
  - [x] `--target wasm32-unknown-unknown`
  - [x] `--target wasm32-unknown-emscripten`
- [x] Run `cargo xtask test` to run tests.
- [x] ~~Add change to `CHANGELOG.md`. See simple instructions inside file.~~

### CODE REVIEW

**Comment by ErichDonGubler** [comment]
I'm not sure why CI doesn't catch this, but I see warnings for these unused fields consistently while running `cargo clippy --workspace --all-targets --all-features` on my M1 Mac workstation.

**Comment by waywardmonkeys** [comment]
CI is running an older version of Rust (currently 1.76).

File: naga/src/front/wgsl/parse/ast.rs [hunk 1]
```diff
@@ -117,33 +117,6 @@ pub struct Function<'a> {
...
-    /// table that maps `Handle<Local>` values to their Naga representations,
-    /// accessed via [`StatementContext::local_table`] and
-    /// [`RuntimeExpressionContext::local_table`]. This table is then consulted when
-    /// lowering subsequent [`Ident`] expressions.
-    ///
-    /// [`LocalDecl`]: StatementKind::LocalDecl
-    /// [`arguments`]: Function::arguments
-    /// [`Ident`]: Expression::Ident
-    /// [`StatementContext::local_table`]: StatementContext::local_table
-    /// [`RuntimeExpressionContext::local_table`]: RuntimeExpressionContext::local_table
```
Comment (teoxoy) [inline naga/src/front/wgsl/parse/ast.rs:144]: We should move this doc comment to the definition of `Local` or `ExpressionContext.locals`

Comment (ErichDonGubler) [reply naga/src/front/wgsl/parse/ast.rs:144]: I've changed the message and contents of 97dba35cfd28e0ce4c31ce6ba7bcaf537462a81f with a push of this PR to 6abce27b9ef7 (viz., 97dba35c). How's it look?

Comment (teoxoy) [reply naga/src/front/wgsl/parse/ast.rs:144]: Looks good, thanks!

**Review by teoxoy — APPROVED** [review]

**Pull request merged**